editor/src/java/com/defold/util/SupportPath.java

```diff
@@ -87,16 +87,14 @@ public static Path getWindowsSupportPath(String applicationName) {
     // linux
 
     private static Path getLinuxSupportPath(String applicationName) {
-        final String userHome = System.getProperty("user.home");
-        if (userHome == null) {
-            return null;
```

Comment:
This code was unnecessarily defensive: the only use of the function (`Editor.getSupportPath()`) expects the returned value to be non-null. Also, `"user.home"` is mostly guaranteed to be non-null, and we expect it to be set to some valid value throughout the editor.